libs/designer/src/lib/ui/panel/templatePanel/templatePanel.tsx

```diff
@@ -59,10 +60,27 @@ export const TemplatePanel = ({ onCreateClick }: { onCreateClick: () => Promise<
     [currentPanelView, templateTitle, templateDescription, manifest?.details, intlText.CREATE_WORKFLOW, intlText.BY_MICROSOFT]
   );
   const onRenderFooterContent = useCallback(
-    () => (selectedTabProps?.footerContent ? <TemplatesPanelFooter {...selectedTabProps?.footerContent} onClose={dismissPanel} /> : null),
+    () =>
+      selectedTabProps?.footerContent ? (
+        <TemplatesPanelFooter
+          {...selectedTabProps?.footerContent}
+          secondaryButtonOnClick={selectedTabProps?.footerContent?.secondaryButtonOnClick ?? dismissPanel}
```

Comment:
I think this should be part of tabprops because if secondary button is not prsent we do not need the click event as well

Reply:
We will render secondary buttons when props are not present; this is when it results in the default case of closing the panel. I can alternatively manually put close for every tab secondary button, but since it's so redundant for a default case, I'm wondering if we do want that